implementation/src/main/java/io/smallrye/config/ConfigValuePropertiesUtils.java

```diff
@@ -0,0 +1,71 @@
+package io.smallrye.config;
+
+import java.lang.invoke.MethodHandles;
+import java.util.function.Consumer;
+
+import org.jboss.logging.Logger;
+
+import io.smallrye.config._private.ConfigLogging;
+
+/**
+ * Class responsible for adding some configuration features to ConfigValueProperties
+ *
+ * Setting system property :
+ * smallrye.config.forbid-duplicate-keys=true (default false)
+ * A ConfigValidationException will be raised if a duplicate key is found in the same config source.
+ *
+ * https://github.com/smallrye/smallrye-config/issues/1269
+ *
+ * NOTE: For backward compatibility, the default behaviour it to avoid duplicate keys check.
+ */
+public class ConfigValuePropertiesUtils {
```

Comment:
If we don't use a configuration to warm / fail and use `.put` semantics, this shouldn't be required.

Reply:
Thanks for the feedback, i changed the warn/fail approach and made it just a warning.

This will only work for properties config sources.

For yaml ones, i didn't find a way to to just warn duplicate keys when reading the yaml config source with SnakeYAML.

This is going to be fine anyway for the projects I'm working on, but at start I wanted to make it a bit more general.

Reply:
Did you check?
https://github.com/smallrye/smallrye-config/blob/1ee672abe1080f3ce212d6f9451f1a3b5a71585f/sources/yaml/src/main/java/io/smallrye/config/source/yaml/YamlConfigSource.java#L66-L98

All the yaml entries are added to a `Map<String, String>`, so instead of `putAll` you could iterate the entries, call `put` and check for the returned object.

Reply:
Hello thanks for the advice @radcortez 

Maybe I'm missing something, but as far as I understand Yaml.loadAll() already returns a single Map where duplicate keys are skipped.

![image](https://github.com/user-attachments/assets/2fa4c4f6-511a-4df6-9681-fcc9a0055bc6)

I think SnakeYAML only allow failing in case of [duplicate](https://www.javadoc.io/doc/org.yaml/snakeyaml/latest/org/yaml/snakeyaml/LoaderOptions.html#isAllowDuplicateKeys()).

**Maybe right now we limit this PR to properties config source?**

NOTE: In ConfigValueProperties the Properties.load() logic is replicated and the check is easy.

About YAML I tried to propose something to SnakeYAML project. : [Option to log duplicate keys](https://bitbucket.org/snakeyaml/snakeyaml/issues/1101/option-to-log-duplicate-keys).

Reply:
Ah, so the YAML code before was to fail and not to remove duplicates? Sorry, I thought it did not remove the duplicates either.

Reply:
Yes, as far I understand SnakeYAML either throws an Exception or removes the duplicate key : 

Here is the [code](https://bitbucket.org/snakeyaml/snakeyaml/src/d19d063fb5e2afb4ff2bb333ec92c6b16acc9479/src/main/java/org/yaml/snakeyaml/constructor/SafeConstructor.java#lines-112) : 

```java
        Integer prevIndex = keys.put(key, i);
        if (prevIndex != null) {
          if (!isAllowDuplicateKeys()) {
            throw new DuplicateKeyException(node.getStartMark(), key,
                tuple.getKeyNode().getStartMark());
          }
          toRemove.add(prevIndex);
        }
```

@radcortez Do you think is acceptable to just log the duplicate keys in property config source? (I think I answered to all other comments too, the current PR content is much simplier than the original one)?